src/Messenger/DI/MessengerExtension.php

```diff
@@ -346,7 +351,7 @@ static function (string $serviceName) use ($builder, $busName): bool {
      *
      * @throws InvalidHandlerService
      */
-    private function getHandledMessageNames(string $serviceName): iterable
+    private function getHandledMessages(string $serviceName): iterable
```

Comment:
Can't we handle both types  here and pass a return iterable of DTOs?

Something like:

```php
final class HandlerDefinition
{
    private string $serviceName;
    private string $methodName;
    
    // ...
}
```

This method would then return `iterable<HandlerDefinition>`.